config/initializers/omniauth.rb

```diff
@@ -0,0 +1,7 @@
+Rails.application.config.middleware.use OmniAuth::Builder do
+  if Site.current
+    facebook_app_id = Setting.get(:facebook, :app_id)
```

Comment:
I'm afraid we can't have something in an initializer that relies on the current Site, since the current site is set on each request. :disappointed: 

Is there a way to configure the OmniAuth middleware per request?